students/chris_lombardi/Lesson05/mailroom_part3.py

```diff
@@ -0,0 +1,105 @@
+#UWPCE PY210
+#Lesson05, Mailroom Part 3
+import sys
+
+donor_log = {} #Log of donors and their respective donation history.
+
+def thank_you():
+    """Accepts donation information and prints a thank you email."""
+    response = input('Please enter a full name or "List":').title()
+
+    #Print the list of donors in the log if requested by the user.
+    while response == 'List':
+        for entry in donor_log:
+            print(entry)
+        response = input('Please enter a full name: ').title()
+
+    try:
+        amount = float(input("Please enter a donation amount:"))
+            #Check if name exists in log and either create new entry or update history.
+        if response not in donor_log:
+            donor_log[response] = [amount]
+        else:
+            donor_log[response] += [amount]
+        #Print a thank you email for the latest donation.
+        print(thankyou_note(response))
+    except ValueError:
+        print("\nNot a valid input.\nPlease try to record donation" +
+              " again using a valid amount.\n\n")
+
+def create_report():
+    """Generate a tabular report of donation history"""
+    header ='\n{:<18}|{:^13}|{:^13}|{:>13}'.format("Donor Name", "Total Given",
+                                                   "Num Gifts", "Average Gift")
+    print(header + '\n' + '-'*len(header))
+    donor_sort = sorted(donor_log, key = sort_key, reverse = True)
+    for entry in donor_sort:
+        total = sum(donor_log.get(entry))
+        num = len(donor_log.get(entry))
+        average = total/num
+        print('{:<18} ${:>12,.2f}{:>13}  ${:>12,.2f}'.format(entry,total,num,average))
+    print('')
+
+def sort_key(entry):
+    return sum(donor_log.get(entry))
+
+def total_donation(name):
+    """Return the sum of donations for a particular donor."""
+    total = 0.0
+    for donation in donor_log.get(name):
+        total += donation
+    return total
```

Comment:
I believe I mentioned this last time, you don't need a loop here - instead use a build in function `sum()`

`sum(donor_log[name])`